### DISCUSSION TITLE
Testing Web Accessibility

### DISCUSSION THREAD

**Discussion by AndreaGriffiths11** [gittogethers/octogatosconf22#133] [Talks in Spanish]
**Speaker: Adrián Bolonio**

"When we develop a new website, we often put a lot of work into the design, to make it beautiful and usable. In other words, we want our website to be effective, efficient and satisfactory for the user. But many times we don't think about the user experience for people with disabilities, including people with age-related disabilities.

Web accessibility (a11y) means that people with disabilities can perceive, understand, navigate and interact with websites and tools, and that they can contribute equally without barriers." (Source: W3C - Web Accessibility Initiative). Our role as frontend developers is to create clear interfaces for people to understand and care about data, regardless of their disabilities or impairments, but what we developers often forget is to make sure that the code we write follows the Web Content Accessibility Guidelines (WCAG), and the only way to achieve this is through testing, either manual or automated.

Automated web accessibility tests can free our QA team from manual testing of every part of our application, but they cannot make our site automatically and magically accessible.

We should use automated web accessibility tests as a step in a larger testing process. Let's not forget that only 20% to 50% of all accessibility issues can be automatically detected.

I will teach some testing tools, libraries and techniques to increase the test coverage of your code with a simple example with a React application."

**Comment by bolonio** [comment]
Aquí tenéis el resumen de mi charla en Español:

> La accesibilidad web no solo es navegación por teclado, contraste de color, o lectores de pantalla. La accesibilidad es un indicador perfecto de la calidad de un sitio web. Cuando un sitio web es accesible, generalmente significa que es inclusivo, usable, ofrece una excelente experiencia de usuario para todos, y además es rápido.
> 
> Nuestro papel como desarrolladores es crear interfaces claras para que las personas entiendan y se preocupen por los datos, independientemente de sus discapacidades, pero a menudo nos olvidamos de asegurarnos de que el código que escribimos cumplan las "Web Content Accessibility Guidelines" (WCAG), y la única forma de lograrlo es mediante test, ya sea manuales o automatizados.
> 
> En esta charla compartiré consejos personales y herramientas útiles a la hora de testear una aplicación o sitio web en busca de vulnerabilidades de accesibilidad web.
